### PR TITLE
feat: add cors in backend

### PR DESCRIPTION
**Pull request by rajat-mehra05**

### CODE REVIEW

File: config/cors.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * The `optionsSuccessStatus` property is a number that tells the browser how to respond to a CORS
+OPTIONS request. 
+ * 
+ * The `credentials` property is a boolean that tells the browser whether to send credentials
+(cookies, for example) with the request. 
+ * 
+ * The `exposedHeaders` property is an array of headers
+ */
+export const corsOptions = {
+  origin: ["http://localhost:3000", "https://socialhq-api.up.railway.app/"],
```
Comment (OmkarK45) [inline config/cors.js:16]: Just make a note here to update this with vercel's URL
you may forget it in future

File: server.js [hunk 2]
```diff
@@ -18,6 +20,9 @@ app.use(express.json());
 require("./models/userModel");
 require("./models/postModel");
 
+//handling cors
+app.use(cors(corsOptions));
```
Comment (OmkarK45) [inline server.js]: put it just after 
```
const app = express()
```

**Review by OmkarK45 — APPROVED** [review]

File: config/cors.js (hunk 1 above)
Comment (rajat-mehra05) [reply config/cors.js:16]: > Just make a note here to update this with vercel's URL you may forget it in future

The local host part, right ?

File: server.js (hunk 2 above)
Comment (rajat-mehra05) [reply server.js]: Okay

File: config/cors.js (hunk 1 above)
Comment (OmkarK45) [reply config/cors.js:16]: > > Just make a note here to update this with vercel's URL you may forget it in future
> 
> The local host part, right ?

its an array of possible hosts that can send requests to your server

so you can just add it like this 
```
['https://myapp.vercel.app', 'http://localhost:3000', 'some other domain .com etc']
```

**Review by OmkarK45 — APPROVED** [review]

**Pull request merged**